### PR TITLE
feat: add onChange prop to searchbar to enable dynamic search

### PR DESCRIPTION
**Pull request by justynoh**
## Problem

Some projects want to implement dynamic searches using design system (i.e. searching on change rather than on keydown Enter), but right now it's a bit clunky. The current workaround is to do this:

```javascript
const Component = () => {
  const handleSearch = (searchterm: string) => { return stuff }
  // Other stuff here
  return (
    <Searchbar 
      onChange = {(e) => handleSearch(e.currentTarget.value)} 
      onSearch = {handleSearch} // Actually not required
    />
  )
}
```
## Solution

Make `onSearch` optional, and add an optional `onChange` prop. This means that users may choose one or the other, and will not be required to implement both.

Since there's a change to the type of `onChange`, this might be breaking for some people.

Feel free to close if unwanted though

### CODE REVIEW

File: react/src/Searchbar/Searchbar.tsx [hunk 1]
```diff
@@ -16,12 +16,18 @@ import {
...
    * @param searchValue value of the search input
    */
-  onSearch: (searchValue: string) => void
+  onSearch?: (searchValue: string) => void
+
+  /**
+   * Function to be invoked when the search input has changed.
+   * @param searchValue value of the search input
+   */
+  onChange?: (searchValue: string) => void
```
Comment (karrui) [inline react/src/Searchbar/Searchbar.tsx:30]: do we need a new function? I think it should be ok enough to make onSearch optional if that is the case. Users of the component can pass in their own `onChange` function to react to changes, no?

**Comment by karrui** [comment]
The workaround seems fine tbh, just need to set the onSearch callback to be optional imho.

**Comment by karrui** [comment]
Closing; reopen if you want to update this PR to allow for an optional `onSearch` prop instead.

File: react/src/Searchbar/Searchbar.tsx (hunk 1 above)
Comment (justynoh) [reply react/src/Searchbar/Searchbar.tsx]: Sounds reasonable, updated in [0c38dc7](https://github.com/opengovsg/design-system/pull/218/commits/0c38dc7ef24adc52c9febcc8e0102375630230e8)

**Review by karrui — APPROVED** [review]

**Pull request merged**